src/features/terms-and-privacy/hooks.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useEffect, useState } from 'react';
+import {
+  getAcceptedLegalDocsVersion,
+  setAcceptedLegalDocsVersion,
+} from './services';
+import { CURRENT_VERSION } from './config';
+
+export const useAcceptDocs = () => {
+  const [accept, setAccept] = useState(false);
+
+  return {
+    accept,
+    setAccept: useCallback(
+      (accepted: boolean) => {
+        setAccept(accepted);
+        setAcceptedLegalDocsVersion(accepted ? CURRENT_VERSION : undefined);
+      },
+      [setAccept]
+    ),
+  };
+};
+
+export const useShowUpdatePrompt = () => {
+  const [showUpdatePrompt, setShowUpdatePrompt] = useState<boolean | undefined>(
```

Comment:
do we need `true`, `false`, and `undefined` here?

Reply:
Yes, without undefined there is a small flickering of wallet and modal.